src/lib/reporter.js

```diff
@@ -162,14 +168,12 @@ class Reporter {
    */
   async sendBugReport(data, token) {
     const logfile = await log.get();
-    const runUrl = this.sendOpenCutsRun(token, data, logfile).catch(
-      () =>
-        "*N/A* <!-- Uploading logs failed. Please add them manually: https://github.com/ubports/ubports-installer#logs -->"
-    );
+    const runUrl = this.sendOpenCutsRun(token, data, logfile).catch(() => null);
+    const logUrl = paste(logfile);
     shell.openExternal(
       `https://github.com/ubports/ubports-installer/issues/new?title=${encodeURIComponent(
         data.title
-      )}&body=${await this.getDebugInfo(data, await runUrl)}`
+      )}&body=${await this.getDebugInfo(data, await runUrl, await logUrl)}`
```

Comment:
Any reason to await here instead of when calling the async function?

Reply:
wouldn't that prevent `sendOpenCutsRun()` and `paste()` from running concurrently?

Reply:
Oh, then it should be `Promise.all` - otherwise there are issues with error handling and whether the requests are actually concurrent is not guaranteed.

Reply:
> issues with error handling

errors in these functions are ignored, so i don't think error handling is an issue.

> whether the requests are actually concurrent is not guaranteed

[concurrency != parallelism](https://stackoverflow.com/a/24684037/7461851). unless i am missing something here, 

[async is based on promises](https://www.freecodecamp.org/news/async-await-and-promises/), using `Promise.all` does not bring a benefit here. the code is more readable like this imo. unless i am missing something about the intricacies of node or i misused syntax smh. but i don't think this is the case.

Reply:
Well if the `logUrl` promise fails before `runUrl` finishes, you get an unhandled error, which if it doesn't already will crash Node at some point in the future.

There is no parallelism in single-threaded applications, only concurrency.

I know async is based on promises, but you don't have a guarantee that the second network request will even fire, since the whole thing can be optimized away to two sequential promises - the intent behind these awaits is not clear.

This is a code smell and it will become a problem at some point.